From 39b5a250e15bc7be71103ce681d67c5fc1a86381 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:44 -0400
Subject: [usb] cdc-wdm: manage_power should always set needs_remote_wakeup

Message-id: <1410887463-20674-213-git-send-email-dzickus@redhat.com>
Patchwork-id: 95170
O-Subject: [RHEL7 PATCH 212/591] usb: cdc-wdm: manage_power should always set needs_remote_wakeup
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 4144bc861ed7934d56f16d2acd808d44af0fcc90
Author: Bjorn Mork <bjorn@mork.no>
Date:   Fri Nov 29 20:17:45 2013 +0100

    usb: cdc-wdm: manage_power should always set needs_remote_wakeup

    Cc: stable <stable@vger.kernel.org>
    Reported-by: Oliver Neukum <oneukum@suse.de>
    Signed-off-by: Bjorn Mork <bjorn@mork.no>
    Acked-by: Oliver Neukum <oneukum@suse.de>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/class/cdc-wdm.c b/drivers/usb/class/cdc-wdm.c
index 4d38759..0b23a86 100644
--- a/drivers/usb/class/cdc-wdm.c
+++ b/drivers/usb/class/cdc-wdm.c
@@ -854,13 +854,11 @@ static int wdm_manage_power(struct usb_interface *intf, int on)
 {
  /* need autopm_get/put here to ensure the usbcore sees the new value */
  int rv = usb_autopm_get_interface(intf);
- if (rv < 0)
-  goto err;
 
  intf->needs_remote_wakeup = on;
- usb_autopm_put_interface(intf);
-err:
- return rv;
+ if (!rv)
+  usb_autopm_put_interface(intf);
+ return 0;
 }
 
 static int wdm_probe(struct usb_interface *intf, const struct usb_device_id *id)
-- 
1.7.1